source/particle/property/crystal_preferred_orientation.cc

```diff
@@ -0,0 +1,1160 @@
+        const Tensor<2,3> spin_tensor = -0.5 *(velocity_gradient_tensor - dealii::transpose(velocity_gradient_tensor));
+
+        return std::pair<std::vector<double>, std::vector<Tensor<2,3> > >(std::vector<double>(n_grains,0.0), std::vector<Tensor<2,3>>(n_grains, spin_tensor));
+      }
+
+      template<int dim>
+      unsigned int
+      CrystalPreferredOrientation<dim>::get_number_of_grains()
+      {
+        return n_grains;
+      }
+
+      template<int dim>
+      unsigned int
+      CrystalPreferredOrientation<dim>::get_number_of_minerals()
+      {
+        return n_minerals;
+      }
+
+
+      template <int dim>
+      void
+      CrystalPreferredOrientation<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Particles");
+          {
+            prm.enter_subsection("Crystal Preferred Orientation");
+            {
+              prm.declare_entry ("Random number seed", "1",
+                                 Patterns::Integer (0),
+                                 "The seed used to generate random numbers. This will make sure that "
+                                 "results are reproducible as long as the problem is run with the "
+                                 "same number of MPI processes. It is implemented as final seed = "
+                                 "user seed + MPI Rank. ");
+
+              prm.declare_entry ("Number of grains per particle", "50",
+                                 Patterns::Integer (0),
+                                 "The number of grains of each different mineral "
+                                 "each particle contains.");
+
+              prm.declare_entry ("Property advection method", "Forward Euler",
+                                 Patterns::Anything(),
+                                 "Options: Forward Euler, Backward Euler, Crank-Nicolson");
+
+              prm.declare_entry ("Property advection tolerance", "1e-10",
+                                 Patterns::Double(0),
+                                 "The Backward Euler and Crank-Nicolson property advection methods involve internal iterations. "
+                                 "This option allows for setting a tolerance. When the norm of tensor new - tensor old is "
+                                 "smaller than this tolerance, the iteration is stopped.");
+
+              prm.declare_entry ("Property advection max iterations", "100",
+                                 Patterns::Integer(0),
+                                 "The Backward Euler and Crank-Nicolson property advection methods involve internal iterations. "
+                                 "This option allows for setting the maximum number of iterations. Note that when the iteration "
+                                 "is ended by the max iteration amount an assert is thrown.");
+
+              prm.declare_entry ("CPO derivatives algorithm", "D-Rex 2004",
```

Comment:
Isnt spin tensor the only one currently working? Make that the default, and remove the mention of D-Rex